src/connectors/youtube.js

```diff
@@ -175,24 +178,42 @@ function getVideoCategory(videoId) {
 		return null;
 	}
 	if (!categoryCache.has(videoId)) {
-		const url = `https://www.googleapis.com/youtube/v3/videos?part=snippet&id=${videoId}&key=${YT_API_KEY}`;
-		fetch(url).then((response) => {
+		fetchCategoryId(videoId).then((category) => {
+			if (category === null) {
+				console.log(`Failed to resolve category for ${videoId}`);
+			}
+			categoryCache.set(videoId, category);
+		});
+	}
+
+	return categoryCache.get(videoId);
+}
+
+async function fetchCategoryId(videoId) {
+	let found = false;
+	for (let key of YT_API_KEYS) {
+		const url = `https://www.googleapis.com/youtube/v3/videos?part=snippet&id=${videoId}&key=${key}`;
+		try {
+			const response = await fetch(url);
+			const data = await response.json();
+
 			if (!response.ok) {
-				throw new Error('Invalid response');
+				console.log(JSON.stringify(data, null, 2));
+				throw new Error(response.statusText);
 			}
-			return response.json();
-		}).then((data) => {
+
 			let category = data.items[0].snippet.categoryId;
 			if (typeof category === 'string') {
-				categoryCache.set(videoId, category);
+				return category;
 			}
-		}).catch((err) => {
-			console.log(`Unable to get category for ${videoId}: ${err.message}`);
-		});
-		return null;
+		} catch (error) {
+			console.log(`Failed fetching category with ${key}`);
+		}
 	}
-	return categoryCache.get(videoId);
 
+	if (!found) {
+		return null;
+	}
```

Comment:
We can just return null value at the end of function, the `found` variable isn't used in the function.